### PR TITLE
V3 - Implement ColumnPairValuesEqual for PandasExecutionEngine

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- JIRA GREAT-56/GREAT-109

### Definition of Done

- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/expectations/metrics/test_core.py [hunk 1]
```diff
@@ -1096,15 +1096,85 @@ def test_table_metric_pd(caplog):
...
+        metrics_to_resolve=(condition_metric,),
+        metrics=metrics,
+    )
+    metrics.update(results)
+
+    assert (
+        results[condition_metric.id][0]
+        .reset_index(drop=True)
+        .equals(pd.Series([False, False, False, True, True]))
+    )
```
Comment (cdkini) [inline tests/expectations/metrics/test_core.py:1131]: Do we want to throw any assertion messages here?

**Review by cdkini — APPROVED** [review]
Awesome work 🎉 

LGTM

File: tests/expectations/metrics/test_core.py (hunk 1 above)
Comment (NathanFarmer) [reply tests/expectations/metrics/test_core.py:1131]: It wouldn't follow the current pattern used elsewhere. I think the developer should be able to follow the logic whereby each test changes based on the `ignore_row_if` argument. If more than one test is failing it is due to some other reason that would require a very generic message.

File: tests/expectations/metrics/test_core.py [hunk 2]
```diff
@@ -1096,15 +1096,85 @@ def test_table_metric_pd(caplog):
...
+    metrics.update(results)
+
+    condition_metric = MetricConfiguration(
         metric_name="column_pair_values.equal.condition",
-        metric_domain_kwargs={"column_A": "a", "column_B": "b"},
-        metric_value_kwargs=dict(),
+        metric_domain_kwargs={
+            "column_A": "a",
+            "column_B": "b",
+            "ignore_row_if": "both_values_are_missing",
```
Comment (alexsherstinsky) [inline tests/expectations/metrics/test_core.py:1115]: @NathanFarmer Any chance we could add another test with `"ignore_row_if": "either_value_is_missing"`?  Thanks!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**